src/client/sandbox/fetch.js

```diff
@@ -0,0 +1,87 @@
+import SandboxBase from './base';
+import nativeMethods from './native-methods';
+import XHR_HEADERS from '../../request-pipeline/xhr/headers';
+import { getProxyUrl, parseProxyUrl } from '../utils/url';
+import { getOrigin, sameOriginCheck, get as getDestLocation } from '../utils/destination-location';
+import { isHeaders, isRequest } from '../utils/dom';
+
+export default class FetchSandbox extends SandboxBase {
+    constructor () {
+        super();
+    }
+
+    static _processRequestInit (init) {
+        var requestDefaults    = new nativeMethods.Request('');
+        var headers            = init.headers || {};
+        var requestCredentials = init.credentials || requestDefaults.credentials;
+
+        if (isHeaders(headers)) {
+            headers.append(XHR_HEADERS.origin, getOrigin());
+            headers.append(XHR_HEADERS.fetchRequestCredentials, requestCredentials);
+        }
+        else {
+            headers[XHR_HEADERS.origin]                  = getOrigin();
+            headers[XHR_HEADERS.fetchRequestCredentials] = requestCredentials;
+        }
+
+        init.headers = headers;
+
+        return init;
+    }
+
+    static _processArguments (args) {
+        var input = args[0];
+        var init  = args[1] || {};
```

Comment:
In specification `Request` parameter is named as `input` and `init` (https://fetch.spec.whatwg.org/#request-class).